src/common/snippets/src/lowered/pass/validate_loops.cpp

```diff
@@ -0,0 +1,84 @@
+//
+
+#include "snippets/lowered/pass/validate_loops.hpp"
+
+#include "snippets/lowered/linear_ir.hpp"
+#include "snippets/lowered/loop_manager.hpp"
+#include "snippets/itt.hpp"
+
+namespace ov {
+namespace snippets {
+namespace lowered {
+namespace pass {
+
+using LoopPort = LinearIR::LoopManager::LoopPort;
+
+ValidateLoops::ValidateLoops() {}
+
+bool ValidateLoops::run(LinearIR& linear_ir) {
+    OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::ValidateLoops")
+    if (linear_ir.empty())
+        return false;
+
+    const auto& loop_manager = linear_ir.get_loop_manager();
+    const auto& loops = loop_manager->get_map();
+
+    // Already validated vectors of Loop IDs
+    std::set<std::vector<size_t>> validated_nested_loops;
+    auto is_already_verified = [&validated_nested_loops](const std::vector<size_t>& ids) {
+        // If `ids` is subsequence of one of loop_ids in `validated_nested_loops` set,
+        // it means that `ids` is already validated too
+        for (const auto& loop : validated_nested_loops) {
+            if (std::search(ids.cbegin(), ids.cend(), loop.cbegin(), loop.cend()) != ids.cend()) {
+                return true;
+            }
+        }
+        return false;
+    };
+
+    std::vector<size_t> dim_indexes;
+
+    auto validate_loop_points = [&loop_manager, &dim_indexes, &validated_nested_loops, &is_already_verified](std::vector<LoopPort>& points) {
+        for (auto& point : points) {
+            const auto expr = point.port->get_expr();
+            const auto loop_ids = expr->get_loop_ids();
+            // If loop_ids of the current port is subsequence of already validated IDs, skip
+            if (is_already_verified(loop_ids))
+                continue;
+
+            dim_indexes.clear();
+            dim_indexes.reserve(loop_ids.size());
+            // Outer Loop -> Inner Loop
+            for (size_t i = 0; i < loop_ids.size(); ++i) {
+                const auto id = loop_ids[i];
+                const auto dim_idx = loop_manager->get_loop_info(id)->dim_idx;
+                if (std::find(dim_indexes.cbegin(), dim_indexes.cend(), dim_idx) == dim_indexes.cend()) {
+                    dim_indexes.push_back(dim_idx);
+                } else {
+                    OPENVINO_ASSERT(*dim_indexes.rbegin() == dim_idx,
+                                    "Incorrect Loop ID configuration: the Loops with splitted dimension should be successively nested");
+                }
+                if (i > 0) {
+                    OPENVINO_ASSERT(loop_manager->get_loop_info(loop_ids[i - 1])->dim_idx >= dim_idx,
+                                    "Incorrect Loop ID configuration: The upper Loop, the upper the corresponding dim_idx");
+                }
+            }
+            validated_nested_loops.insert(loop_ids);
+        }
+    };
+
+    for (const auto& pair : loops) {
+        const auto& loop_info = pair.second;
+        validate_loop_points(loop_info->entry_points);
```

Comment:
Minor: there is no such thing as `loop_point` anymore. Please rename to `loop_port`

Reply:
Updated, thanks
7ec2d5186c46e0e181494de5b589dea2d7cfae1d